cdap-ui/app/cdap/components/PipelineDetails/PipelineDetailsTopPanel/PipelineDetailsButtons/PipelineStopButton/index.js

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+*/
+
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import {MyProgramApi} from 'api/program';
+import IconSVG from 'components/IconSVG';
+import Alert from 'components/Alert';
+import {getCurrentNamespace} from 'services/NamespaceStore';
+import {GLOBALS} from 'services/global-constants';
+import PipelineStopPopover from 'components/PipelineDetails/PipelineDetailsTopPanel/PipelineDetailsButtons/PipelineStopButton/PipelineStopPopover';
+
+export default class PipelineStopButton extends Component {
+  static propTypes = {
+    isBatch: PropTypes.bool,
+    pipelineName: PropTypes.string,
+    currentRun: PropTypes.object,
+    runs: PropTypes.array
+  }
+
+  state = {
+    loading: false,
+    stopError: null,
+    disabled: false,
+    runningRuns: []
+  };
+
+  componentWillReceiveProps(nextProps) {
+    let runningRuns = nextProps.runs.filter(run => run.status === 'RUNNING');
+    this.setState({
```

Comment:
nit: We might want to check if the props have changed. Could avoid a setState and a re-render.